### PR TITLE
some better typing

### PR DESCRIPTION
**Pull request by rouseabout**
Fixes for making https://github.com/geohot/tinygrad/commit/084bdd0f49541c43a3ec4635b670ec58f855e7b3 run with CPU=1.

There may be a better way to make Tensor.neg work for integer types. Open to ideas.

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/cstyle.py                 4               1
tinygrad/runtime/ops_cpu.py                1               1
tinygrad/tensor.py                         1               1
------------------------------------------------------------
total                                      6               3
------------------------------------------------------------
lines added in the tinygrad folder: 3
```

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -471,7 +471,7 @@ def cos(self): return ((math.pi/2)-self).sin()
   def tan(self): return self.sin() / self.cos()
   # ***** math functions (unary) *****
 
-  def __neg__(self): return 0.0-self
+  def __neg__(self): return Tensor.zeros(*self.shape, dtype=self.dtype)-self
```
Comment (geohot) [inline tinygrad/tensor.py:474]: You definitely shouldn't have to do this.

File: tinygrad/codegen/cstyle.py [hunk 2]
```diff
@@ -121,7 +121,10 @@ def kk(s): kernel.append("  "*depth+s)
       if bufs[args.i] is not None and isinstance(bufs[args.i].realized, RawConst):
         assert newvar.ltype == LocalTypes.float, "const can't be float4"
         # nan? inf?
-        val = f"{bufs[args.i].realized._buf}" + ("f" if not dtypes.is_int(bufs[args.i].dtype) else "")
+        if dtypes.is_int(bufs[args.i].dtype) or dtypes.is_float(bufs[args.i].dtype):
+            val = f"{bufs[args.i].realized._buf}" + ("f" if dtypes.is_float(bufs[args.i].dtype) else "")
```
Comment (geohot) [inline tinygrad/codegen/cstyle.py:125]: Wrong indentation.

File: tinygrad/tensor.py [hunk 3]
```diff
@@ -475,7 +475,7 @@ def cos(self): return ((math.pi/2)-self).sin()
   def tan(self): return self.sin() / self.cos()
   # ***** math functions (unary) *****
 
-  def __neg__(self): return 0.0-self
+  def __neg__(self): return Tensor(0, dtype=self.dtype if not isinstance(self.dtype,ImageDType) else dtypes.float32)-self
```
Comment (geohot) [inline tinygrad/tensor.py:478]: This should be fixed elsewhere.

**Comment by geohot** [comment]
Won't merge. It's a hack for only __neg__

**Pull request closed without merging**